service/src/main/java/uk/gov/hmcts/reform/fpl/service/CreateC21OrderService.java

```diff
@@ -107,12 +107,22 @@ private String getCourtName(String courtName) {
             .collect(toList());
     }
 
-    private String mostRecentUploadedC21DocumentUrl(final List<Element<C21Order>> c21Orders) {
-        return getLast(c21Orders.stream()
+    public String mostRecentUploadedC21DocumentUrl(final List<Element<C21Order>> c21Orders) {
+        String mostRecentUploadedC21DocumentUrl = getLast(c21Orders.stream()
             .filter(Objects::nonNull)
             .map(Element::getValue)
             .filter(Objects::nonNull)
             .collect(toList()))
             .getDocument().getBinaryUrl();
+
+        final String documentHostUrl = documentConfiguration.getDocumentHostUrl();
+
+        try {
+            URL url = new URL(mostRecentUploadedC21DocumentUrl);
+            mostRecentUploadedC21DocumentUrl = documentHostUrl + url.getPath();
+        } catch (MalformedURLException e) {
+            log.error("mostRecentUploadedC21DocumentUrl url incorrect.", e);
+        }
+        return mostRecentUploadedC21DocumentUrl;
```

Comment:
Could we extract to sdeparatee method? A bit messy. Could allow some mapping. 

Can you swap from URL to URI? UIR is a bit more lightweight class. 

Not sure about returning original URL if it fail to parse, probably would be better to return some placeholder text? It would be invalid URL so not sure.
Are you sure the "e" contains the URL for the debugging purposes?